relayer/src/cli/programs/relay.ts

```diff
@@ -59,10 +59,35 @@ export class RelayerProgram {
   }
 
   async run(): Promise<void> {
-    await this.l2RelayerTo.prepare();
-    const l1TransactionHash = await this.l2RelayerFrom.relayTxToL1(this.l2TransactionHash);
-    if (l1TransactionHash) {
-      await this.l2RelayerTo.finalize(l1TransactionHash);
+    const relayStepCompleted = await this.l2RelayerFrom.relayTxToL1Step.isCompleted(
+      this.l2TransactionHash,
+    );
+    let l1TransactionHash: string;
+
+    if (relayStepCompleted) {
+      l1TransactionHash = await this.l2RelayerFrom.relayTxToL1Step.recoverL1TransactionHash(
+        this.l2TransactionHash,
+      );
+    } else {
+      if (this.l2RelayerTo.prepareStep) {
+        const prepareStepCompleted = await this.l2RelayerTo.prepareStep.isCompleted();
+
+        if (prepareStepCompleted === false) {
```

Comment:
We could now change this to `if(!prepareStepCompleted)`

Reply:
Are true! thanks!